// @bpanel/bpanel-utils/dist/clients.js
/*!
 * clients.js - client extensions of bclient to connect to bpanel app server
 * Copyright (c) 2018, Bcoin Devs (MIT License).
 * https://github.com/bcoin-org/bpanel-utils
 */

'use strict';

exports.__esModule = true;
exports.bwalletClient = exports.bpanelClient = exports.walletPath = exports.nodePath = exports.ssl = exports.bpanelPort = undefined;

var _bclient = require('bclient');

var _options = options(),
    bpanelPort = _options.bpanelPort,
    ssl = _options.ssl,
    nodePath = _options.nodePath,
    walletPath = _options.walletPath;

exports.bpanelPort = bpanelPort;
exports.ssl = ssl;
exports.nodePath = nodePath;
exports.walletPath = walletPath;


var nodeClient = null;
var walletClient = null;

function bpanelClient() {
  if (!nodeClient) nodeClient = new _bclient.NodeClient({ port: bpanelPort, path: nodePath, ssl: ssl });

  return nodeClient;
}

function bwalletClient() {
  if (!walletClient) walletClient = new _bclient.WalletClient({
    port: bpanelPort,
    path: walletPath,
    ssl: ssl
  });

  return walletClient;
}

function options() {
  // bpanel endpoints
  var nodePath = '/bcoin';
  var walletPath = '/bwallet';
  // determine the port and ssl usage
  var protocol = window.location.protocol;
  var port = window.location.port;
  var ssl = false;
  // use https and http ports when the window doesn't render them
  if (port === '') protocol === 'https:' ? port = '443' : port = '80';
  if (protocol === 'https:') ssl = true;
  return {
    bpanelPort: parseInt(port),
    ssl: ssl,
    nodePath: nodePath,
    walletPath: walletPath
  };
}

exports.bpanelClient = bpanelClient;
exports.bwalletClient = bwalletClient;
